src/pandapipes/pipeflow.py

```diff
@@ -68,136 +67,147 @@ def pipeflow(net, sol_vec=None, **kwargs):
     init_all_result_tables(net)
 
     create_lookups(net)
-    node_pit, branch_pit = initialize_pit(net)
-    if len(node_pit) == 0:
-        logger.warning("There are no nodes defined. "
-                       "You need at least one node! "
-                       "Without any nodes, you are not able to conduct a pipeflow!")
-        return
+    initialize_pit(net)
 
     calculation_mode = get_net_option(net, "mode")
-    calculate_hydraulics = calculation_mode in ["hydraulics", "all"]
-    calculate_heat = calculation_mode in ["heat", "all"]
+    calculate_hydraulics = calculation_mode in ["hydraulics", 'sequential']
+    calculate_heat = calculation_mode in ["heat", 'sequential']
+    calculate_bidrect = calculation_mode == "bidirectional"
 
     # cannot be moved to calculate_hydraulics as the active node/branch hydraulics lookup is also required to
     # determine the active node/branch heat transfer lookup
-    identify_active_nodes_branches(net, branch_pit, node_pit)
-
-    if calculation_mode == "heat":
-        if not net.user_pf_options["hyd_flag"]:
-            raise UserWarning("Converged flag not set. Make sure that hydraulic calculation "
-                              "results are available.")
-        else:
-            net["_pit"]["node"][:, PINIT] = sol_vec[:len(node_pit)]
-            net["_pit"]["branch"][:, MDOTINIT] = sol_vec[len(node_pit):]
-
-    if calculate_hydraulics:
-        reduce_pit(net, node_pit, branch_pit, mode="hydraulics")
-        hydraulics(net)
-        if not net.converged:
-            raise PipeflowNotConverged("The hydraulic calculation did not converge to a solution.")
-        extract_results_active_pit(net, mode="hydraulics")
-
-    if calculate_heat:
-        node_pit, branch_pit = net["_pit"]["node"], net["_pit"]["branch"]
-        identify_active_nodes_branches(net, branch_pit, node_pit, False)
-        reduce_pit(net, node_pit, branch_pit, mode="heat_transfer")
-        heat_transfer(net)
-        if not net.converged:
-            raise PipeflowNotConverged("The heat transfer calculation did not converge to a "
-                                       "solution.")
-        extract_results_active_pit(net, mode="heat_transfer")
-    elif not calculate_hydraulics:
+    identify_active_nodes_branches(net)
+
+    if calculation_mode == 'heat': use_given_hydraulic_results(net, sol_vec)
```

Comment:
I would like to avoid one-line if conditions. They are not marked by codacy, but I find it less readable.